src/agent/communicator/src/http_socket.cpp

```diff
@@ -0,0 +1,161 @@
+#include "http_socket.hpp"
+
+#include "http_client_utils.hpp"
+#include <ihttp_socket.hpp>
+#include <logger.hpp>
+
+#include <boost/asio.hpp>
+#include <boost/asio/experimental/awaitable_operators.hpp>
+#include <boost/beast.hpp>
+#include <boost/system/error_code.hpp>
+
+#include <exception>
+#include <string>
+
+namespace http_client
+{
+    HttpSocket::HttpSocket(const boost::asio::any_io_executor& ioContext)
+        : m_socket(ioContext)
+    {
+    }
+
+    void HttpSocket::Connect(const boost::asio::ip::tcp::resolver::results_type& endpoints,
+                             boost::system::error_code& ec)
+    {
+        try
+        {
+            boost::asio::connect(m_socket, endpoints, ec);
+        }
+        catch (const std::exception& e)
+        {
+            LogDebug("Exception thrown: {}", e.what());
+        }
+    }
+
+    boost::asio::awaitable<void> HttpSocket::AsyncConnect(
+        const boost::asio::ip::tcp::resolver:: // NOLINT(cppcoreguidelines-avoid-reference-coroutine-parameters)
+        results_type& endpoints,
+        boost::system::error_code& ec) // NOLINT(cppcoreguidelines-avoid-reference-coroutine-parameters)
+    {
+        using namespace boost::asio::experimental::awaitable_operators;
+
+        auto timer = std::make_shared<boost::asio::steady_timer>(co_await boost::asio::this_coro::executor);
+        timer->expires_after(std::chrono::seconds(http_client_utils::TIMEOUT_SECONDS));
+
+        auto result = std::make_shared<boost::system::error_code>();
+        auto taskCompleted = std::make_shared<bool>(false);
+
+        try
+        {
+            co_await (http_client_utils::TimerTask(timer, result, taskCompleted) ||
+                      http_client_utils::SocketConnectTask(m_socket, endpoints, result, taskCompleted));
+            if (!result)
```

Comment:
This looks like a pointer check for null, and not a check on the actual error code. I've seen this check in some other places in the PR...

Reply:
Fixed